tests/test_graphnet_backend.py

```diff
@@ -0,0 +1,442 @@
+def test_covid_test_positive_result(database, setup_backend_database):
+    setup_backend_database(
+        Patients(Patient_ID=1),
+        PracticeRegistrations(
+            Patient_ID=1, StartDate="2001-01-01", EndDate="2026-06-26"
+        ),
+        CovidTestResults(
+            Patient_ID=1,
+            SpecimenDate="2020-05-05",
+            positive_result=True,
+        ),
+        backend="graphnet",
+    )
+
+    class Cohort:
+        date = (
+            table("covid_test_results")
+            .filter(positive_result=True)
+            .earliest()
+            .get("date")
+        )
+
+    assert extract(Cohort, GraphnetBackend, database) == [
+        dict(patient_id=1, date=date(2020, 5, 5))
+    ]
+
+
+@pytest.mark.integration
+def test_covid_test_negative_result(database, setup_backend_database):
+    setup_backend_database(
+        Patients(Patient_ID=1),
+        PracticeRegistrations(
+            Patient_ID=1, StartDate="2001-01-01", EndDate="2026-06-26"
+        ),
+        CovidTestResults(
+            Patient_ID=1,
+            SpecimenDate="2020-05-05",
+            positive_result=False,
+        ),
+        backend="graphnet",
+    )
+
+    class Cohort:
+        date = (
+            table("covid_test_results")
+            .filter(positive_result=False)
+            .earliest()
+            .get("date")
+        )
+
+    assert extract(Cohort, GraphnetBackend, database) == [
+        dict(patient_id=1, date=date(2020, 5, 5))
+    ]
+
+
+@pytest.mark.integration
+def test_patients_table(database, setup_backend_database):
+    setup_backend_database(
+        Patients(Patient_ID=1, Sex="F", DateOfBirth="1950-01-01"),
+        PracticeRegistrations(
+            Patient_ID=1, StartDate="2001-01-01", EndDate="2026-06-26"
+        ),
+        backend="graphnet",
+    )
+
+    class Cohort:
+        _patients = table("patients").first_by("patient_id")
+        sex = _patients.get("sex")
+        dob = _patients.get("date_of_birth")
+
+    assert extract(Cohort, GraphnetBackend, database) == [
+        dict(patient_id=1, sex="F", dob=date(1950, 1, 1))
+    ]
+
+
+@pytest.mark.integration
+def test_hospitalization_table_returns_admission_date_and_code(
+    database, setup_backend_database
+):
+    setup_backend_database(
+        *patient(
+            1,
+            "M",
+            "1990-1-1",
+            registration("2001-01-01", "2026-06-26"),
+            hospitalization(admit_date="2020-12-12", code="xyz"),
+        ),
+        backend="graphnet",
+    )
+
+    class Cohort:
+        _hospitalization = table("hospitalizations").first_by("patient_id")
+        admission = _hospitalization.get("date")
+        code = _hospitalization.get("code")
+
+    assert extract(Cohort, GraphnetBackend, database) == [
+        dict(patient_id=1, admission=date(2020, 12, 12), code="xyz")
+    ]
+
+
+""" Removing hospitalization tests - code parsing doesn't occur in the graphnet backend, so these won't work
```

Comment:
Could you remove these commented out tests that we don't need? Otherwise it all LGTM now :tada: